src/main/java/impl/image/ImageConverterImpl.java

```diff
@@ -0,0 +1,171 @@
+
+import java.awt.*;
+import java.lang.reflect.Array;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.function.Function;
+
+public class ImageConverterImpl implements ImageConverter {
+
+    private Function<Color, Integer> colorValueGetter;
+    private Function<Integer, Color> colorValueSetter;
+    private boolean valueIsPrivate = true;
+
+    private int set8Bits(int value, int pos) {
+        return value << (24 - pos);
+    }
+
+    private int get8Bits(int value, int pos) {
+        return (value >> (24 - pos)) & 0xFF;
+    }
+
+    private int deconstructColor(Color pixel) {
+        return set8Bits(pixel.getAlpha(), 0) | set8Bits(pixel.getRed(), 8) |
+                set8Bits(pixel.getGreen(), 16) | set8Bits(pixel.getBlue(), 24);
+    }
+
+    private Color constructColor(int pixel) {
+        return new Color(get8Bits(pixel, 8), get8Bits(pixel, 16), get8Bits(pixel, 24), get8Bits(pixel, 0));
+    }
+
+    /**
+     * Works sorta like a one-time generated property with backing field
+     * Gets a function that gets value field of a Color object.
+     * If reflection access was OK - it's direct getting value of the private field of a given Color object
+     * The fallback just does what's obvious approach - sets various bit parts of a number.
+     */
+    private Function<Color, Integer> getColorValueGetterFunction() {
+        if (colorValueGetter == null) {
+            try {
+                var field = Color.class.getDeclaredField("value");
+
+                if (valueIsPrivate) {
+//                  Yeah, the fun part about java is that if you get the field, `setAccessible` will never throw
+                    field.setAccessible(true);
+                    valueIsPrivate = false;
+                }
+
+                colorValueGetter = (Color color) -> {
+                    try {
+                        return (int) field.get(color);
+                    } catch (IllegalAccessException e) {
+                        e.printStackTrace();
+//            Weird fallback, for some reason there's no way here to use it as a functor and apply like w'd really want.
+                        return deconstructColor(color);
+                    }
+                };
+            } catch (NoSuchFieldException e) {
+                colorValueGetter = this::deconstructColor;
+            }
+        }
+        return colorValueGetter;
+    }
+
+    /**
+     * Works sorta like a one-time generated property with backing field
+     * Gets a function that sets value field of a Color object.
+     * If reflection access was OK - it's direct setting to the private field of a freshly created Color object
+     * The fallback just does what's obvious approach - gets different bit parts and constructs proper object
+     */
+    private Function<Integer, Color> getColorValueSetterFunction() {
+        if (colorValueSetter == null) {
+
+            try {
+                var field = Color.class.getDeclaredField("value");
```

Comment:
Look closely at `Color` class's methods and constructors

Reply:
There's https://github.com/maiksaray/java-is-2020/tree/task-1-no-reflection where I did look) For some reason I did not see it from the start)